absolute albumArtURI + some minors

cederigo opened the pull request.

jishi commented:
Regarding album art, I have a user case where I actually want the relative URL. I use the relative URL to the browser in my web controller, to proxy the actual request through node.js. I then cache that response in order to avoid hitting the players again for the same image. It could of course be rewritten but I also use the hash of the URL to identify cache files, which would differ if I were to include a random IP in it as well.

Regarding state, I see now that you have replicated the transport-state emit. This is probably fine to change, my main concern was that perhaps the state object had a reference to the actual player in charge of the state change, which might have been used by some to interact with the system. I could see that as an improvement overall, meaning a native state object which also has a reference to the actual player involved which would allow someone to interact on an event with that player directly without having to find it using UUID. Maybe also include the convertToSimple() method on that state object. However, that would be a breaking change and probably not a good idea to change now. If you revert the group state I will merge the other changes. 

Regarding queueInUse, the different variants that I can remember right now is:

queue (x-rincon-queue)
line-in (x-rincon-stream)
radio (this is tricky, it can differ a lot depending on source), maybe differentiate between tunein and custom radios and pandora/deezer/spotify radio of possible.
line-in spdif (for playbar, x-rincon-htastream I think, could be the same as regular line-in)
optionally, grouped, but then it will not be a coordinator state (x-rincon:) and might not be very useful, it's easy enough to check anyway (compare player with player.coordinator).

the word source might be mistaken for different music services, which wouldn't be the case here. maybe playbackType?

cederigo commented:
Sorry for my late answer. I was on vacation. Without any computer ;-)

I did various updates based on your feedback:

**albumArtURI**
I leave "albumArtURI" as is and include "absoluteAlbumArtURI" too.

**state events**
Reverted the changes to the group events

**queueInUse**
For now I will use state.currentTrack.type to figure out if the queue is in use or not. When it says "track" I assume that its from the queue. Therefore I reverted the changes I made completely. 

Maybe tackle this problem more in depth in another pull request?

Sorry, I accidentally deleted my branch. So made a new pull request #34
